scheme/common/arm/handlers.go

```diff
@@ -10,47 +10,68 @@ import (
 	"github.com/veraison/services/proto"
 	"github.com/veraison/services/scheme/common"
+	realm "github.com/veraison/services/scheme/common/cca/realm"
 )
 
 type SwAttr struct {
-	ImplID           []byte `cca:"CCA_SSD_PLATFORM.impl-id" psa:"PSA_IOT.impl-id" parcca:"PARSEC_CCA.impl-id"`
-	Model            string `cca:"CCA_SSD_PLATFORM.hw-model" psa:"PSA_IOT.hw-model" parcca:"PARSEC_CCA.hw-model"`
-	Vendor           string `cca:"CCA_SSD_PLATFORM.hw-vendor" psa:"PSA_IOT.hw-vendor" parcca:"PARSEC_CCA.hw-vendor"`
-	MeasDesc         string `cca:"CCA_SSD_PLATFORM.measurement-desc" psa:"PSA_IOT.measurement-desc" parcca:"PARSEC_CCA.measurement-desc"`
-	MeasurementType  string `cca:"CCA_SSD_PLATFORM.measurement-type" psa:"PSA_IOT.measurement-type" parcca:"PARSEC_CCA.measurement-type"`
-	MeasurementValue []byte `cca:"CCA_SSD_PLATFORM.measurement-value" psa:"PSA_IOT.measurement-value" parcca:"PARSEC_CCA.measurement-value"`
-	SignerID         []byte `cca:"CCA_SSD_PLATFORM.signer-id" psa:"PSA_IOT.signer-id" parcca:"PARSEC_CCA.signer-id"`
-	Version          string `cca:"CCA_SSD_PLATFORM.version" psa:"PSA_IOT.version" parcca:"PARSEC_CCA.version"`
+	ImplID           []byte `json:"impl-id"`
+	Model            string `json:"hw-model"`
+	Vendor           string `json:"hw-vendor"`
+	MeasDesc         string `json:"measurement-desc"`
+	MeasurementType  string `json:"measurement-type"`
+	MeasurementValue []byte `json:"measurement-value"`
+	SignerID         []byte `json:"signer-id"`
+	Version          string `json:"version"`
 }
 
 type TaAttr struct {
-	Model    string `cca:"CCA_SSD_PLATFORM.hw-model" psa:"PSA_IOT.hw-model" parcca:"PARSEC_CCA.hw-model"`
-	Vendor   string `cca:"CCA_SSD_PLATFORM.hw-vendor" psa:"PSA_IOT.hw-vendor" parcca:"PARSEC_CCA.hw-vendor"`
-	VerifKey string `cca:"CCA_SSD_PLATFORM.iak-pub" psa:"PSA_IOT.iak-pub" parcca:"PARSEC_CCA.iak-pub"`
-	ImplID   []byte `cca:"CCA_SSD_PLATFORM.impl-id" psa:"PSA_IOT.impl-id" parcca:"PARSEC_CCA.impl-id"`
-	InstID   string `cca:"CCA_SSD_PLATFORM.inst-id" psa:"PSA_IOT.inst-id" parcca:"PARSEC_CCA.inst-id"`
+	Model    string `json:"hw-model"`
+	Vendor   string `json:"hw-vendor"`
+	VerifKey string `json:"iak-pub"`
+	ImplID   []byte `json:"impl-id"`
+	InstID   string `json:"inst-id"`
 }
 
 type CcaPlatformCfg struct {
-	ImplID []byte `cca:"CCA_SSD_PLATFORM.impl-id" parcca:"PARSEC_CCA.impl-id"`
-	Model  string `cca:"CCA_SSD_PLATFORM.hw-model" parcca:"PARSEC_CCA.hw-model"`
-	Vendor string `cca:"CCA_SSD_PLATFORM.hw-vendor" parcca:"PARSEC_CCA.hw-vendor"`
-	Label  string `cca:"CCA_SSD_PLATFORM.platform-config-label" parcca:"PARSEC_CCA.platform-config-label"`
-	Value  []byte `cca:"CCA_SSD_PLATFORM.platform-config-id" parcca:"PARSEC_CCA.platform-config-id"`
+	ImplID []byte `json:"impl-id"`
+	Model  string `json:"hw-model"`
+	Vendor string `json:"hw-vendor"`
+	Label  string `json:"platform-config-label"`
+	Value  []byte `json:"platform-config-id"`
 }
 
 func SynthKeysFromRefValue(scheme string, tenantID string,
```

Comment:
There is no point in having this method in common code, as it is just  a switch on scheme, and so is not actually common code.

Instead  implement a `SynthKeysFromRefValue` for each of the shcems bleow that justs calls the relevant  function.

Reply:
@setrofim : May be you are seeing the old code ? Can you please check, again, this method no longer exists! It is just in the squashing sequence, you may be seeing this ?
Only method exposed is: `SynthKeysForPlatform` & `GetPlatformReferenceIDs` which does not have any scheme specific code changes!

Reply:
I was reviewing one commit at a time, it's possilbe this got replaced by a later commit..

Reply:
Sure, this has been addressed and I have rechecked it again today!